### ISSUE TITLE
[Bug]: Automigration addReact fails

### ISSUE THREAD

**Issue by AasmundEndresen** [storybookjs/storybook#23505]
### Describe the bug

When running the automigration everything hangs - and nothing moves on.  If, as requested by someone here I answer 'no' to this REQUIRED automigration for 'addon-docs' - I get an error that '@babel/preset-react' is missing. If I install this peer-dependency that isn't mentioned anywhere and try to manually add react and react-dom as suggested in the migration documentation, I get stuck in an infinite loop that can't even be exited without killing the terminal.

It is literally IMPOSSIBLE for me to upgrade to Storybook 7, and this have kept me occupied for over a month now trying to fix - and is costing our company ALOT of money to have resources tied up in. Any response would be great...

**Comment by shilman** [comment]
Do you a have a reproduction repo you can share? If not, can you create one? Go to https://storybook.new or see [repro docs](https://storybook.js.org/docs/react/contribute/how-to-reproduce).

We've successfully tested the 7.0 migration process on a wide variety of projects, so there's no way for us to know what's going wrong without being able to see the error happening and play around with it.

**Comment by AasmundEndresen** [comment]
> Do you a have a reproduction repo you can share? If not, can you create one? Go to https://storybook.new or see [repro docs](https://storybook.js.org/docs/react/contribute/how-to-reproduce).
> 
> We've successfully tested the 7.0 migration process on a wide variety of projects, so there's no way for us to know what's going wrong without being able to see the error happening and play around with it.

Creating a reproduction would be difficult as we have a large project and have been using Storybook and evolving with it from the start - our repository is also private because of the nature of the organization - I understand this makes it more difficult to play around with and debug but its the nature of the situation.

Could we perhaps arrange a session/meeting over Teams and I could present it more clearly and answer questions directly.  We've been having trouble with the migration process since we first attempted it in January, and we also get issues if we try to tear down everything Storybook related and initialising Storybook from scratch.

Because of security policies we also use a private registry and need to refactor our storybook extensively to get it to work with 7.0 because of the breaking changes which also makes the debugging process on our end increasingly frustrating and difficult as I simply cannot test anything on the same day that you have made new versions, because the sub-dependencies under storybook are being published after the main version and thus don't necessarily exist in our private registry at the same version that @latest is pointing to.

I also ran into an issue where it seemed '@babel/preset-typescript' was a peerDependency but wasn't mentioned in the docs, if I ignore the react/react-dom installation above and try to build storybook - so while you have successfully tested the migration process its clearly lacking its share of documentation and legible error messages. As when resolving that issue again, I get this error when trying to load stories: "Error: Didn't find 'my-component--default' in CSF file, this is unexpected at StoryStore.storyFromCSFFile".
